app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb

```diff
@@ -368,9 +354,16 @@ def find_activation_keys
       activation_keys
     end
 
-    def find_foreman_host(organization)
-      Host.where(:name => params[:facts]['network.hostname'],
-                 :organization_id => organization.id).first if params[:facts]
+    def find_or_create_foreman_host(organization)
+      hosts = ::Host.where(:name => params[:facts]['network.hostname'])
+      if hosts.empty? #no host exists
+        ::Host::Managed.new_from_rhsm_params(system_params, organization, Location.default_location)
```

Comment:
Without the subscription aspect, would this function still exist/be
used/make sense?

On Thu, Oct 1, 2015 at 1:09 PM, Justin Sherrill <notifications@github.com>
wrote:

> In app/controllers/katello/api/rhsm/candlepin_proxies_controller.rb
> <https://github.com/Katello/katello/pull/5495#discussion_r40939349>:
>
> > @@ -368,9 +354,16 @@ def find_activation_keys
> >        activation_keys
> >      end
> >
> > -    def find_foreman_host(organization)
> > -      Host.where(:name => params[:facts]['network.hostname'],
> > -                 :organization_id => organization.id).first if params[:facts]
> > +    def find_or_create_foreman_host(organization)
> > +      hosts = ::Host.where(:name => params[:facts]['network.hostname'])
> > +      if hosts.empty? #no host exists
> > +        ::Host::Managed.new_from_rhsm_params(system_params, organization, Location.default_location)
>
> I can move it there, but a Host is being created not a
> subscription-aspect, so it seemed a little odd to stick it in the
> subscription_aspect . thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5495/files#r40939349>.
>



-- 
Eric D. Helms
Red Hat Engineering
Ph.D. Student - North Carolina State University